storage-provider/server/src/storage.rs

```diff
@@ -186,91 +192,94 @@ async fn upload(
                 (StatusCode::INTERNAL_SERVER_ERROR, err.to_string())
             })?
     } else {
-        // Read the request body into a CAR archive
+        // For direct uploads, convert the request body into a stream.
         let body_reader = StreamReader::new(
             request
                 .into_body()
                 .into_data_stream()
                 .map_err(|err| io::Error::new(io::ErrorKind::Other, err)),
         );
-        stream_contents_to_car(state.car_piece_storage_dir.clone().as_ref(), body_reader)
+        stream_contents_to_car(state.car_piece_storage_dir.as_ref(), body_reader)
             .await
             .map_err(|err| {
                 tracing::error!(%err, "failed to store file into CAR archive");
                 (StatusCode::INTERNAL_SERVER_ERROR, err.to_string())
             })?
     };
-    tracing::debug!("generated cid: {file_cid}");
 
-    // NOTE(@jmg-duarte,03/10/2024): Maybe we should just register the file in RocksDB and keep a
-    // background process that vacuums the disk as necessary to simplify error handling here
+    tracing::debug!("generated cid: {file_cid}");
 
+    // Open the CAR file to check its size.
     let (_, file_path) = content_path(&state.car_piece_storage_dir, file_cid);
     let file = File::open(&file_path).await.map_err(|err| {
         tracing::error!(%err, path = %file_path.display(), "failed to open file");
         (StatusCode::INTERNAL_SERVER_ERROR, err.to_string())
     })?;
     let file_size = file
         .metadata()
-        .map_ok(|metadata| metadata.len())
         .await
+        .map(|m| m.len())
         .map_err(|err| (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()))?;
 
-    // Check the piece size first since it's the cheap check
+    // Check that the piece size matches the proposal.
     let piece_size = PaddedPieceSize::from_arbitrary_size(file_size);
-    if !(proposed_deal.piece_size == *piece_size) {
+    if proposed_deal.piece_size != *piece_size {
         tracing::trace!(
             expected = proposed_deal.piece_size,
             actual = *piece_size,
             "piece size does not match the proposal piece size"
         );
-
-        // Not handling the error since there's little to be done here...
-        let _ = tokio::fs::remove_file(&file_path).await.inspect_err(
-            |err| tracing::error!(%err, path = %file_path.display(), "failed to delete file"),
-        );
-
+        let _ = tokio::fs::remove_file(&file_path).await;
         return Err((
             StatusCode::BAD_REQUEST,
             "piece size does not match proposal".to_string(),
         ));
     }
 
     let piece_path = file_path.clone();
-    // Calculate the piece commitment in the blocking thread pool since `calculate_piece_commitment`
-    // is CPU intensive — i.e. blocking — potentially improvement is to move this completely out of
-    // the tokio runtime into an OS thread
```

Comment:
Recurring theme: why are you removing comments that explain the context behind the decisions?